src/iceberg/manifest_entry.cc

```diff
@@ -0,0 +1,159 @@
+        SchemaField::MakeRequired(122, std::string(MapType::kValueName),
+                                  std::make_shared<LongType>())));
+const SchemaField DataFile::NAN_VALUE_COUNTS = SchemaField::MakeOptional(
+    137, "nan_value_counts",
+    std::make_shared<MapType>(
+        SchemaField::MakeRequired(138, std::string(MapType::kKeyName),
+                                  std::make_shared<IntType>()),
+        SchemaField::MakeRequired(139, std::string(MapType::kValueName),
+                                  std::make_shared<LongType>())));
+const SchemaField DataFile::LOWER_BOUNDS = SchemaField::MakeOptional(
+    125, "lower_bounds",
+    std::make_shared<MapType>(
+        SchemaField::MakeRequired(126, std::string(MapType::kKeyName),
+                                  std::make_shared<IntType>()),
+        SchemaField::MakeRequired(127, std::string(MapType::kValueName),
+                                  std::make_shared<BinaryType>())));
+const SchemaField DataFile::UPPER_BOUNDS = SchemaField::MakeOptional(
+    128, "upper_bounds",
+    std::make_shared<MapType>(
+        SchemaField::MakeRequired(129, std::string(MapType::kKeyName),
+                                  std::make_shared<IntType>()),
+        SchemaField::MakeRequired(130, std::string(MapType::kValueName),
+                                  std::make_shared<BinaryType>())));
+const SchemaField DataFile::KEY_METADATA =
+    SchemaField::MakeOptional(131, "key_metadata", std::make_shared<BinaryType>());
+const SchemaField DataFile::SPLIT_OFFSETS = SchemaField::MakeOptional(
+    132, "split_offsets",
+    std::make_shared<ListType>(SchemaField::MakeRequired(
+        133, std::string(ListType::kElementName), std::make_shared<LongType>())));
+const SchemaField DataFile::EQUALITY_IDS = SchemaField::MakeOptional(
+    135, "equality_ids",
+    std::make_shared<ListType>(SchemaField::MakeRequired(
+        136, std::string(ListType::kElementName), std::make_shared<IntType>())));
+const SchemaField DataFile::SORT_ORDER_ID =
+    SchemaField::MakeOptional(140, "sort_order_id", std::make_shared<IntType>());
+const SchemaField DataFile::FIRST_ROW_ID =
+    SchemaField::MakeOptional(142, "first_row_id", std::make_shared<LongType>());
+const SchemaField DataFile::REFERENCED_DATA_FILE = SchemaField::MakeOptional(
+    143, "referenced_data_file", std::make_shared<StringType>());
+const SchemaField DataFile::CONTENT_OFFSET =
+    SchemaField::MakeOptional(144, "content_offset", std::make_shared<LongType>());
+const SchemaField DataFile::CONTENT_SIZE =
+    SchemaField::MakeOptional(145, "content_size_in_bytes", std::make_shared<LongType>());
+
+StructType DataFile::GetType(StructType partition_type) {
+  std::vector<SchemaField> fields;
+
+  fields.push_back(CONTENT);
+  fields.push_back(FILE_PATH);
+  fields.push_back(FILE_FORMAT);
+  fields.push_back(SchemaField::MakeRequired(
+      102, "partition", std::make_shared<StructType>(partition_type)));
```

Comment:
Why not directly accept a `std::shared_ptr<StructType>` as the input parameter?

Reply:
agree, will do.